tests/test_operational_limits.py

```diff
@@ -0,0 +1,220 @@
+class BasicActivity(plugins.HasRequestWindowPluginActivity, model.BasicActivity):
+    """
+    Model a simple activity with operational limits.
+
+    An instance of the `BasicActivity` class holds the required
+    activity parameters for the model to be processed. The class sets up
+    the framework for defining activities. The `model.BasicActivity`
+    parent class provides the main part of the framework, the
+    `plugins.HasRequestWindowPluginActivity` adds the feature of
+    operational limits to the activity.
+
+    Parameters
+    ----------
+        env: simpy.Environment
+            A SimPy simulation environment.
+        registry: dict
+            A dictionary in which the activities are stored.
+        name: str
+            A descriptive name for storing the activity in the log.
+        duration: int
+            The number of seconds it takes to execute the activity.
+        offshore_environment: plugins.OffshoreEnvironment
+            An instance of the OffshoreEnvironment.
+        limit_expr: Callable
+            A Python function expressing the operational limits in
+            terms of offshore environment parameters. See an example
+            below.
+    """
+
+    def __init__(self, **kwargs):
+        """Class constructor."""
+        super().__init__(**kwargs)
+
+
+# -------------------------------------------------------------------------------------!
+# Define the operational limits
+def limit_expression(hs, tp):
+    """
+    Define the operational limits as a python function.
+
+    Using Python functions, we can set the conditions for which
+    the operations may not be performed. The function should
+    return a `True` value to indicate that the operational limit
+    has been exceeded for a given combination of input
+    parameters.
+
+    """
+    # if Hs OR Tp is greater than 0.50, the limit is exceeded.
+    return (hs > 0.5) | (tp > 0.5)
+
+
+# -------------------------------------------------------------------------------------!
+def create_sample_data():
+    """Create sample data for the test simulation."""
+    # Generate sample dates
+    dates = np.arange(
+        start=datetime.datetime(2021, 1, 1),
+        stop=datetime.datetime(2021, 1, 2),
+        step=datetime.timedelta(hours=1),
+    )
+
+    # Generate sample data and export to *.csv
+    hs_values = np.random.random(size=(len(dates)))
+    tp_values = np.random.random(size=(len(dates)))
+
+    hs_df = pd.DataFrame({"date": dates, "hs": hs_values}).set_index("date")
+    tp_df = pd.DataFrame({"date": dates, "tp": tp_values}).set_index("date")
+
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    hs_df.to_csv(current_dir + "/data/unit_hs.csv")
+    tp_df.to_csv(current_dir + "/data/unit_tp.csv")
+
+    return current_dir
+
+
+# -------------------------------------------------------------------------------------!
+if __name__ == "__main__":
+    exec()
```

Comment:
Please use the propper pytest sytaxt. This is not nessescary. Pytest already searches for tests if it has certain prefixes. (for methods test_ and the classes Tests)